fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/export/checkpoint/ExportCheckpointAlgorithm.java

```diff
@@ -25,11 +31,20 @@
 
     private final static Logger logger = Logger.getLogger(ExportCheckpointAlgorithm.class.getName());
 
+    @Inject
+    JobContext jobCtx;
+
     @Inject
     StepContext stepCtx;
 
+    final boolean isFileExport;
+
     public ExportCheckpointAlgorithm() {
-        // No Operation
+        JobOperator jobOperator = BatchRuntime.getJobOperator();
+        JobExecution jobExecution = jobOperator.getJobExecution(jobCtx.getExecutionId());
+        BatchContextAdapter contextAdapter = new BatchContextAdapter(jobExecution.getJobParameters());
+        BulkDataContext ctx = contextAdapter.getStepContextForSystemChunkWriter();
+        isFileExport = "file".equals(ConfigurationFactory.getInstance().getStorageProviderType(ctx.getSource()));
```

Comment:
Don't do this here. It's unreliable with CDI

Reply:
ok, I moved it to beginCheckpoint() and hid it behind a null check (so it should only run the first time through).